src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -213,19 +213,23 @@ abstract class UnCurry extends InfoTransform
         // Expand the function body into an anonymous class
         gen.expandFunction(localTyper)(fun, inConstructorFlag)
       } else {
+        val mustExpand = mustExpandFunction(fun)
+        val newNamePrefix = if (mustExpand) nme.DELAMBDAFY_LAMBDA_CLASS_NAME else nme.ANON_FUN_NAME
```

Comment:
so if we expand to an anon class the method is called `$lambda`, otherwise `$anonfun` - why that distinction?